### DISCUSSION TITLE
dynamic topic kafka with condition

### DISCUSSION THREAD

**Discussion by taquanghung17051999** [vectordotdev/vector#15275] [Q&A] [answered]
I have a question that:
I'm source from http and sink to kafka

I want when I send log to http have path is: / => log is sent to kafka with topic is named:
Ex:
/amg.a => topicA
/amg.b => topicB
/amg.c => topicC

How to do this ?

**Comment by spencergilbert** [comment, marked as answer]
👋 
Hi, @taquanghung17051999. The [topic](https://vector.dev/docs/reference/configuration/sinks/kafka/#topic) configuration is _templateable_ - you can see example usage on the linked page [here](https://vector.dev/docs/reference/configuration/template-syntax/).

**Reply by taquanghung17051999** [reply]
So I have to create a field to store name of topic by transform and if command, right ?
Finally, at variable topic (sink kafka) I set:
topic = "{{field_topic}}"

**Reply by spencergilbert** [reply]
Correct, or if there is an existing field (for example [kubernetes.pod_namespace](https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#pod_annotation_fields.pod_namespace)) it would be

```toml
topic = "{{ .kubernetes.pod_namespace }}"
```

**Reply by dev-jimbo** [reply]
does anyone know if this syntax also applies to sinks?
like for a table name to be dynamic??

[sinks.db]
table = "{{ .kubernetes.pod_namespace}}"

**Reply by spencergilbert** [reply]
Hi @dev-jimbo - it's available across all components, in the documentation it'll be labeled with a `template` tag:

<img width="721" alt="image" src="https://user-images.githubusercontent.com/11903873/207606191-ebd0d96e-9e3e-42df-b758-4333d38ece87.png">

**Reply by dev-jimbo** [reply]
thanks @spencergilbert  - but i noticed its not available for everything, for example, sinks.clickhouse for table name attribute does not have template icon, but you can consume from multiple topics

**Reply by spencergilbert** [reply]
Yes, it's not a feature for every option - it's documented where implemented, if it's not it's not currently supported. If you have a need for expanded template support you can open an issue requesting the feature and discuss the reasoning to add it.

For example, there's an open issue for clickhouse: https://github.com/vectordotdev/vector/issues/14283
